Cargo.toml

```diff
@@ -12,8 +12,8 @@ readme = "README.md"
 edition = "2018"
 
 [dependencies]
+derive_is_enum_variant = "0.1.1"
 failure = "0.1.3"
```

Comment:
@fairingrey Should we be injecting `failure` as a dependency for library? If it is used, other users need to pull it as well. IIRC most libraries does not use failure.

Reply:
Not sure, you'll have to ask @yoshuawuyts. It was already present when I started working on this PR :sweat_smile:

I would probably remark that there might be a better way to do error handling, but I don't have any strong opinions on it. FWIW, per failure's best practices we do implement `Fail` for our Error type so it's easier to work with: https://docs.rs/failure/0.1.5/failure/